### ISSUE TITLE
Running the same task app in parallel via CTR on CF, droplets fail to create occassionally

### ISSUE THREAD

**Issue by cppwfs** [spring-cloud/spring-cloud-dataflow#4448]
To reproduce this execute the following composed task `t1: timestamp && <t2: timestamp && t3: timestamp || t4: timestamp ||t5: timestamp> && t6: timestamp"`

DataFlow deletes  artifacts after task launch to prevent the maven repository filling up the disk, however if another task is being deployed during the delete the droplet is not created and thus the composed task fails to run. 
The current workaround is to set `spring.cloud.dataflow.task.platform.cloudfoundry.accounts[default].deployment.auto-delete-maven-artifacts` to false.   But is there a better way so we don't use up all the disk.

**Comment by jvalkeal** [comment]
If was thinking if we could do a semi naive locking with a cf deployer using java's `StampedLock` if this setting is enabled. Kinda using this type of logic if there's multiple parallel deploy operations:

1. As issue is within one jvm we don't need to think about fully shared access.
2. Deployer would always acquire read lock from StampedLock
3. When it wants to clean stuff, it could do naive attempt to try writelock
4. If it can get writelock, delete file and release

With this most likely last parallel deploy would get write lock and able to delete. One getting read lock might get blocked for short time when delete is happening.
